pkg/syncer/spec/spec_process.go

```diff
@@ -256,9 +256,10 @@ func (c *Controller) ensureSyncerFinalizer(ctx context.Context, gvr schema.Group
 		namespace := upstreamObjCopy.GetNamespace()
 		logicalCluster := logicalcluster.From(upstreamObjCopy)
 
-		upstreamFinalizers = append(upstreamFinalizers, shared.SyncerFinalizerNamePrefix+c.syncTargetName)
+		upstreamFinalizers = append(upstreamFinalizers, shared.SyncerFinalizerNamePrefix+c.syncTargetKey)
 		upstreamObjCopy.SetFinalizers(upstreamFinalizers)
-		if _, err := c.upstreamClient.Cluster(logicalCluster).Resource(gvr).Namespace(namespace).Update(ctx, upstreamObjCopy, metav1.UpdateOptions{}); err != nil {
+		if _, err := c.upstreamClient.Cluster(logicalCluster).Resource(gvr).Namespace(namespace).Patch(ctx, name, types.MergePatchType, []byte(`{"metadata":{"finalizers":["`+shared.SyncerFinalizerNamePrefix+c.syncTargetKey+`"]}}`), metav1.PatchOptions{}); err != nil {
```

Comment:
this doesn't work I think. A merge patch with a slice will replace the slice. You have to append.